### DISCUSSION TITLE
Field of type Date inside identifier leads to No Single View Generated error

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#477] [Q&A] [answered]
### What product are you having troubles with?

Fast Data

### What Console version are you using?

v12.0.2

### Description

Hi!
I have a single view that has a field of type Date as Identifier.
So in the Projection Changes schema I setup identifier adding this field.

What is happening is that the SVTG is producing correctly the sv-trigger message. The SVC (version 6.4.0) is throwing an error like this one:
`Single View generation threw an error
    identifier: {
      "field1": "01",
      "field2": null,
      "fieldDate": "2024-04-22T00:00:00.000Z",
    }
    kafkaInfo: {}
    err: {
      "type": "SingleViewGenerationError",
      "message": "No Single View record generated",
      "stack":
          Error: No Single View record generated
              at /home/node/app/node_modules/@mia-platform-internal/single-view-creator-lib/dist/lib/single-view-creation.js:340:15
              at Generator.next (<anonymous>)
              at fulfilled (/home/node/app/node_modules/@mia-platform-internal/single-view-creator-lib/dist/lib/single-view-creation.js:25:24)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    }
`

### Actual Outcome

The SVC is not able to find the record of the base projection so is throwing an error.

### Expected Outcome

Correct generation of the single view

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @elmalakomar 

When a date field is sent over a kafka topic, it will be represented as a string.
So, in the SVC, you'll need to parse the identifier contained in the trigger message before starting the aggregation process.

This is possible by the usage of [the advanced section of the Fast Data No Code](https://docs.mia-platform.eu/docs/fast_data/configuration/config_maps/aggregation#use-the-no-code): there, you can use the [identifierQueryMapping field](https://docs.mia-platform.eu/docs/fast_data/configuration/config_maps/aggregation#changing-the-query-that-finds-the-projection-based-on-their-identifier) of the base projection dependency to use a custom function that parses the date accordingly.

In the `SV_CONFIG` object you'll have something like this:

```json
{
  "version": "1.3.0",
  "config": {
    "SV_CONFIG": {
      "dependencies": {
         // ... some dependencies
        "PROJECTION_NAME": {
          "on": "_identifier",
          "identifierQueryMapping": "__fromFile__[castIdentifier.js]"
        }
        // ... some dependencies
      },
      "mapping": {
        // ...here there is your mapping
       }
    }
  }
}
```

Then, you'll have to create a custom function named `castIdentifier.js` which will do the following:

```js
/** 
 * converts the "fieldDate" field of the identifier into a date object
 */
module.exports = async function(logger, clientMongo, dependenciesMap) {

    // access the _identifier object from the dependenciesMap, is the identifier of the sv trigger message
    const identifier = dependenciesMap.get("_identifier")
    return {
      ...identifier
      // here I override fieldDate
      fieldDate: new Date(identifier.fieldDate)
    }
}
```

Let me know how it goes! 

@albertotessarotto lets keep track of this example since can be useful to improve the aggregation configuration

**Reply by elmalakomar** [reply]
This solution worked smoothly! thanks
